src/main/java/seedu/clinic/ui/AutoCompleteTextField.java

```diff
@@ -0,0 +1,107 @@
+package seedu.clinic.ui;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javafx.geometry.Side;
+import javafx.scene.control.ContextMenu;
+import javafx.scene.control.CustomMenuItem;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import seedu.clinic.logic.commands.AddCommand;
+import seedu.clinic.logic.commands.AssignMacroCommand;
+import seedu.clinic.logic.commands.ClearCommand;
+import seedu.clinic.logic.commands.DeleteCommand;
+import seedu.clinic.logic.commands.EditCommand;
+import seedu.clinic.logic.commands.ExitCommand;
+import seedu.clinic.logic.commands.FindCommand;
+import seedu.clinic.logic.commands.HelpCommand;
+import seedu.clinic.logic.commands.ListCommand;
+import seedu.clinic.logic.commands.RemoveMacroCommand;
+import seedu.clinic.logic.commands.UpdateCommand;
+import seedu.clinic.logic.commands.ViewCommand;
+
+
+/**
+ * A TextField with added implementation of "autocomplete" functionality.
+ * Entries are based on Command_Usage.
+ */
+public class AutoCompleteTextField extends TextField {
+    private static final SortedSet<String> entries = new TreeSet<>();
+    private ContextMenu popUpEntries;
+
+    /**
+     * Constructor for autocomplete.
+     */
+    public AutoCompleteTextField() {
+        super();
+        this.setEntries();
+        popUpEntries = new ContextMenu();
+        textProperty().addListener((observableValue, s, s2) -> {
+            if (getText().length() == 0) {
+                popUpEntries.hide();
+            } else {
+                popUpEntries.hide();
+                LinkedList<String> searchResult = new LinkedList<>();
```

Comment:
Probably good to include some comment to describe the logic of this part since both TreeSet and LinkedList are used here.

Reply:
Good idea, will add in more comments